core/shared/src/main/scala/com/dwolla/consul/ConsulServiceDiscoveryAlg.scala

```diff
@@ -145,25 +148,42 @@ object ConsulServiceDiscoveryAlg {
                                                             initialConsulIndex: Option[ConsulIndex],
                                                             consulBase: Uri,
                                                             longPollTimeout: FiniteDuration,
-                                                            client: Client[F]): Resource[F, F[Vector[Uri.Authority]]] =
+                                                            client: Client[F],
+                                                            entryPoint: EntryPoint[F],
+                                                           )
+                                                           (implicit L: Local[F, Span[F]]): Resource[F, F[Vector[Uri.Authority]]] =
     Stream.unfoldEval(initialConsulIndex) { maybeIndex =>
-      // since this is a background task, it doesn't make sense to attach it to the trace that initially started it
-      import natchez.Trace.Implicits.noop
```

Comment:
The `noop` `Trace[F]` previously passed to `lookup` prevented the span creation immediately within `lookup` (e.g. the span named `com.dwolla.consul.ConsulServiceDiscoveryAlg.lookup`) but the issue is observable when using a `Client[F]` that has been wrapped in the Natchez middleware from natchez-http4s. Assuming it's using a `Local[F, Span[F]]`-based MTL `Trace[F]`, the span referenced by the Natchez middleware is not the noop span, but the current value of the `Local` instance, meaning the middleware creates a new span and sends that trace's kernel as part of the request being made.